lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  type AddonInfo,
+  type Compatibility,
+  DescribeAddonVersionsCommand,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import * as awsEksAddonVersioning from '../../versioning/aws-eks-addon';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { EksAddonsFilter } from './schema';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  override readonly defaultVersioning = awsEksAddonVersioning.id;
+  override readonly caching = true;
+  private readonly clients: Record<string, EKSClient> = {};
+
+  constructor() {
+    super(AwsEKSAddonDataSource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${AwsEKSAddonDataSource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `getReleases:${packageName}`,
+  })
+  async getReleases({
+    packageName: serializedFilter,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const res = EksAddonsFilter.safeParse(serializedFilter);
+    if (!res.success) {
+      logger.warn(
+        { err: res.error, serializedFilter },
+        'Error parsing eks-addons config.',
+      );
+      return null;
+    }
+
+    const filter = res.data;
+
+    const cmd = new DescribeAddonVersionsCommand({
+      kubernetesVersion: filter?.kubernetesVersion,
+      addonName: filter?.addonName,
+      maxResults: 1,
+    });
+    const response = await this.getClient(filter).send(cmd);
+    const addons: AddonInfo[] = response.addons ?? [];
+    return {
+      releases: addons
+        .flatMap((addon) => {
+          return addon.addonVersions;
+        })
+        .map((versionInfo) => ({
+          version: versionInfo?.addonVersion ?? '',
+          default:
+            versionInfo?.compatibilities?.some(
+              (comp: Compatibility): boolean | undefined => comp.defaultVersion,
+            ) ?? false,
+          compatibleWith: versionInfo?.compatibilities?.flatMap(
+            (comp: Compatibility): string | undefined => comp.clusterVersion,
+          ),
+        }))
```

Comment:
```suggestion
        .filter(is.nonEmptyObject)
        .map((versionInfo) => ({
          version: versionInfo.addonVersion ?? '',
          default:
            versionInfo.compatibilities?.some(
              (comp: Compatibility): boolean | undefined => comp.defaultVersion,
            ) ?? false,
          compatibleWith: versionInfo.compatibilities?.flatMap(
            (comp: Compatibility): string | undefined => comp.clusterVersion,
          ),
        }))
```

Reply:
you forgot this

Reply:
Hi @viceice 

The proposed changes breaks the code

![Screenshot 2025-03-20 at 07 08 37](https://github.com/user-attachments/assets/2a2f1b44-50b2-45cd-beac-8bb36373156c)

![Screenshot 2025-03-20 at 07 11 23](https://github.com/user-attachments/assets/9d6735f9-96fd-47a3-8f98-abfdc34717be)

I've added instead of ` .filter(is.nonEmptyObject)`

```
.filter((versionInfo): versionInfo is { addonVersion: string; compatibilities: Compatibility[] } => is.nonEmptyObject(versionInfo))
```

Does it make sense?